diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bf5317..88b4600 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1054,16 +1054,16 @@ INSTALL(FILES ${man3} DESTINATION man/man3)
 INSTALL(FILES ${html} DESTINATION share/doc/pcre2/html)
 
 IF(MSVC AND INSTALL_MSVC_PDB)
- INSTALL(FILES ${PROJECT_BINARY_DIR}/pcre2-8.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-16.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-32.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-posix.pdb
+ INSTALL(FILES ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-8.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-16.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-32.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-posix.pdb
          DESTINATION bin
          CONFIGURATIONS RelWithDebInfo)
- INSTALL(FILES ${PROJECT_BINARY_DIR}/pcre2-8d.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-16d.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-32d.pdb
-               ${PROJECT_BINARY_DIR}/pcre2-posixd.pdb
+ INSTALL(FILES ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-8d.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-16d.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-32d.pdb
+               ${PROJECT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX_CXX}pcre2-posixd.pdb
          DESTINATION bin
          CONFIGURATIONS Debug)
 ENDIF(MSVC AND INSTALL_MSVC_PDB)
